### DISCUSSION TITLE
Telescope can't find `neoclip` extension

### DISCUSSION THREAD

**Discussion by latipun7** [LunarVim/LunarVim#2916] [Q&A] [answered]
I've installed [neoclip](https://github.com/AckslD/nvim-neoclip.lua), but the telescope can't find its [extension](https://github.com/AckslD/nvim-neoclip.lua/tree/main/lua/telescope/_extensions). I can't run `:Telescope neoclip`
Please help.

Here's how I call the plugin:
```lua
lvim.plugins = {
  {
    "AckslD/nvim-neoclip.lua",
    config = function()
      require("neoclip").setup()
    end,
  },
}
```
after `:PackerSync`, and exit lvim, then enter lvim again, I still can't start neoclip with `:Telescope neoclip` even tho the plugin is successfully installed, and `:PackerStatus` shows that neoclip is loaded.

Please help, at least how to debug what causing this issue. Thank you very much🙏

**Comment by rebuilt** [comment, marked as answer]
Can you try going through the troubleshooting steps on that plugins readme first:   https://github.com/AckslD/nvim-neoclip.lua#troubleshooting

**Reply by latipun7** [reply]
Thank you for your quick reply.
That page said that packer doesn't need to run `:lua require('telescope').load_extension('neoclip')` first but it still can't run `:Telescope neoclip`. Then I just run that command, but error:

<details>
<summary>Error log</summary>

```log
E5108: Error executing lua .../start/telescope.nvim/lua/telescope/_extensions/init.lua:10: 'neoclip' extension doesn't exist or isn't i
nstalled: module 'telescope._extensions.neoclip' not found:
        no field package.preload['telescope._extensions.neoclip']
        no file './telescope/_extensions/neoclip.lua'
        no file '/usr/share/luajit-2.1.0-beta3/telescope/_extensions/neoclip.lua'
        no file '/usr/local/share/lua/5.1/telescope/_extensions/neoclip.lua'
        no file '/usr/local/share/lua/5.1/telescope/_extensions/neoclip/init.lua'
        no file '/usr/share/lua/5.1/telescope/_extensions/neoclip.lua'
        no file '/usr/share/lua/5.1/telescope/_extensions/neoclip/init.lua'
        no file '/home/latipun/.cache/lvim/packer_hererocks/2.1.0-beta3/share/lua/5.1/telescope/_extensions/neoclip.lua'
        no file '/home/latipun/.cache/lvim/packer_hererocks/2.1.0-beta3/share/lua/5.1/telescope/_extensions/neoclip/init.lua'
        no file '/home/latipun/.cache/lvim/packer_hererocks/2.1.0-beta3/lib/luarocks/rocks-5.1/telescope/_extensions/neoclip.lua'
        no file '/home/latipun/.cache/lvim/packer_hererocks/2.1.0-beta3/lib/luarocks/rocks-5.1/telescope/_extensions/neoclip/init.lua'
        no file './telescope/_extensions/neoclip.so'
        no file '/usr/local/lib/lua/5.1/telescope/_extensions/neoclip.so'
        no file '/usr/lib/lua/5.1/telescope/_extensions/neoclip.so'
        no file '/usr/local/lib/lua/5.1/loadall.so'
        no file '/home/latipun/.cache/lvim/packer_hererocks/2.1.0-beta3/lib/lua/5.1/telescope/_extensions/neoclip.so'
        no file './telescope.so'
        no file '/usr/local/lib/lua/5.1/telescope.so'
        no file '/usr/lib/lua/5.1/telescope.so'
        no file '/usr/local/lib/lua/5.1/loadall.so'
        no file '/home/latipun/.cache/lvim/packer_hererocks/2.1.0-beta3/lib/lua/5.1/telescope.so'
stack traceback:
        [C]: in function 'error'
        .../start/telescope.nvim/lua/telescope/_extensions/init.lua:10: in function 'load_extension'
        .../start/telescope.nvim/lua/telescope/_extensions/init.lua:62: in function 'load_extension'
        [string ":lua"]:1: in main chunk
```

</details>

Then I tried to restart lvim and run `:lua require('telescope').extensions.neoclip.default()` instead. And it works now with this command.

But the problem is when I tried to make neoclip optional, all that 3 commands didn't work.
Here's how I make neoclip optional:
```lua
lvim.plugins = {
  {
    "AckslD/nvim-neoclip.lua",
    keys = "<leader>y",
    config = function()
      require("neoclip").setup()
    end,
  },
}
```
PackerSync and PackerCompile.
Now the `:lua require('telescope').extensions.neoclip.default()` give the same error message as I run `:lua require('telescope').load_extension('neoclip')`.

**Reply by rebuilt** [reply]
Don't lazy load this extension.  Setting the `keys` option in packer means the plugin will `ONLY` be loaded after those keys are pressed.   This means neoclip will definitely not be loaded when you pass the command `:lua require('telescope').extensions.neoclip.default()`.   

You might try defining your own command and then passing that command as the trigger for packer to load the plugin, but then you ALSO need the plugin to be loaded when you press your leader key.  I'm not sure if packer picks the first lazy load option or considers ALL lazy load options.  That's beyond my familiarity with Packer. 

Either way, it's simpler to just not lazy load this plugin.  

I tested it without the lazy loading statement and it seems to work as long as you trigger it with `:lua require('telescope').extensions.neoclip.default()`

Additionally, I'll say that if you're just interested in the history of the things you've yanked, you can pull up the registers by typing `:reg` or just hitting the quotation mark to see the history.  You can type `"ap` to paste what's in the `a` register.

**Reply by rebuilt** [reply]
Telescope also provides the builtin `:Telescope registers`

**Reply by latipun7** [reply]
Thank you. I'll stick with starting this plugin and call it with `:lua require('telescope').extensions.neoclip.default()` for now.
